app/controllers/recipes_controller.rb

```diff
@@ -90,4 +87,14 @@ def unlike
     render 'like'
   end
 
+  private
+
+  def recipe
+    @recipe ||= Recipe.find(params[:id])
+  end
+
+  def recipe_params
+    params.require(:recipe).permit(:name, :photo, :category_id, :num_persons, :cook_time, :directions,
+      new_recipe_ingredients_attributes: [:optional, :value, :value_type, :name, :comment])
```

Comment:
Line is too long. [91/80]<br>Align the parameters of a method call if they span more than one line.